ansible/roles/env_hacks/tasks/main.yml

```diff
@@ -25,21 +25,36 @@
+          - name: Use canary apiserver
+            shell: "{{ oc_cmd }} patch -n service-catalog deployment/apiserver -p '{\"spec\":{\"template\": {\"spec\": {\"containers\": [{\"name\":\"apiserver\",\"image\":\"{{ coalmine_apiserver }}\", \"command\": [\"/opt/services/apiserver\"], \"args\": {{ arglist.stdout }} }]}}}}'"
 
-      - name: Get current controller-manager args and update them
-        shell: "{{ oc_cmd }} export -n service-catalog deployment controller-manager -o json | python -c 'import json, sys; args=json.load(sys.stdin)[\"spec\"][\"template\"][\"spec\"][\"containers\"][0][\"args\"]; args.remove(\"--service-catalog-api-server-url\"); args.remove(\"http://$(APISERVER_SERVICE_HOST):$(APISERVER_SERVICE_PORT)\"); args.append(\"--service-catalog-api-server-url\"); args.append(\"https://$(APISERVER_SERVICE_HOST):$(APISERVER_PORT_443_TCP_PORT)\"); args.append(\"--service-catalog-insecure-skip-verify\"); print json.dumps(args)'"
-        register: arglist
+          - name: Get current controller-manager args and update them
+            shell: "{{ oc_cmd }} export -n service-catalog deployment controller-manager -o json | python -c 'import json, sys; args=json.load(sys.stdin)[\"spec\"][\"template\"][\"spec\"][\"containers\"][0][\"args\"]; args.remove(\"--service-catalog-api-server-url\"); args.remove(\"http://$(APISERVER_SERVICE_HOST):$(APISERVER_SERVICE_PORT)\"); args.append(\"--service-catalog-api-server-url\"); args.append(\"https://$(APISERVER_SERVICE_HOST):$(APISERVER_PORT_443_TCP_PORT)\"); args.append(\"--service-catalog-insecure-skip-verify\"); print json.dumps(args)'"
+            register: arglist
 
-      - name: Use canary controller-manager
-        shell: "{{ oc_cmd }} patch -n service-catalog deployment/controller-manager -p '{\"spec\":{\"template\": {\"spec\": {\"containers\": [{\"name\":\"controller-manager\",\"image\":\"{{ coalmine_controller }}\", \"command\": [\"/opt/services/controller-manager\"], \"args\": {{ arglist.stdout }} }]}}}}'"
+          - name: Use canary controller-manager
+            shell: "{{ oc_cmd }} patch -n service-catalog deployment/controller-manager -p '{\"spec\":{\"template\": {\"spec\": {\"containers\": [{\"name\":\"controller-manager\",\"image\":\"{{ coalmine_controller }}\", \"command\": [\"/opt/services/controller-manager\"], \"args\": {{ arglist.stdout }} }]}}}}'"
+        when: openshift_client_version == '3.6'
+
+      - name: Update service-catalog (OpenShift 3.7)
+        block:
+        - name: Workaround canary permissions issue
+          shell: "{{ oc_cmd }} adm policy add-cluster-role-to-user cluster-admin system:anonymous"
+
+        - name: Use canary apiserver
+          shell: "{{ oc_cmd }} patch -n service-catalog deployment/apiserver -p '{\"spec\":{\"template\": {\"spec\": {\"containers\": [{\"name\":\"apiserver\",\"image\":\"{{ coalmine_apiserver }}\", \"command\": [\"/opt/services/apiserver\"] }]}}}}'"
```

Comment:
-n kube-service-catalog worked for me. I'm not sure if this is a (very) recent change? I didn't have this issue the other day when we hit this, I don't believe.

Reply:
I don't remember it being in `kube-service-catalog`. I'll go another round with the latest and see.